core/src/test/scala/org/broadinstitute/dsde/rawls/mock/MockWorkspaceManagerDAO.scala

```diff
@@ -66,7 +66,7 @@ class MockWorkspaceManagerDAO extends WorkspaceManagerDAO {
   }
 
   override def deleteDataReference(workspaceId: UUID, referenceId: UUID, accessToken: OAuth2BearerToken): Unit = {
-    references.getOrElse((workspaceId, referenceId), throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, "Not found")))
-    references -= ((workspaceId, referenceId))
+    if (references.contains(workspaceId, referenceId))
```

Comment:
A test for deleting a non-existent snapshot already exists. See https://github.com/broadinstitute/rawls/blob/develop/core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/SnapshotApiServiceSpec.scala#L383

Updating the mock here to match current behavior.